lib/client.js

```diff
@@ -1311,16 +1311,33 @@ Client.prototype.info = function (request, host, policy, callback) {
  * })
  */
 Client.prototype.infoAny = function (request, policy, callback) {
+  let req = request
+  let pol = policy
+
   if (typeof request === 'function') {
     callback = request
     request = null
-    policy = null
```

Comment:
If the first parameter is a function, then that must be the callback function. Any parameters passed _after_ the callback function should be ignore, or – better yet – a TypeError should be raised.

Reply:
A required parameter cannot follow an optional parameter but I see many places we use that way Is that allowed in nodejs?

Reply:
never mind, i see only in couple places like that

Reply:
Callback function is optional. Which examples do you see where a required param follows an optional one?